src/avl_tree/set.rs

```diff
@@ -0,0 +1,400 @@
+                    left_ptrs.push(&mut **next_left_node);
+                }
+                next_tree = &mut next_left_node.left;
+            }
+
+            let parent_leftest_node = unsafe { &mut *left_ptrs.pop().unwrap() };
+
+            let mut leftest_node = parent_leftest_node.left.take().unwrap();
+
+            // Taken node is now filled with leftest value
+            let inner_value = replace(&mut target_node.value, leftest_node.value);
+
+            // Leftest node is now the right child of former leftest,
+            // because leftest has no left child and if right child is none, then thats the end of this tree
+            replace(&mut parent_leftest_node.left, leftest_node.right.take());
+
+            // Start at the bottom with updating
+            parent_leftest_node.update_height();
+            parent_leftest_node.rebalance();
+
+            // Up to the children of target
+            // Rev because into iter starts at the first inserted item, we need the last inserted first
+            for node_ptr in left_ptrs.into_iter().rev() {
+                let node = unsafe { &mut *node_ptr };
+                node.update_height();
+                node.rebalance();
+            }
+
+            // At last of course target node to update
+            target_node.update_height();
+            target_node.rebalance();
+
+            inner_value
+        }
+    }
+
+    pub fn remove(&mut self, value: &T) -> bool {
+        self.take(value).is_some()
+    }
+
+    pub fn contains(&self, value: &T) -> bool {
+        self.get(value).is_some()
+    }
+
+    pub fn get(&self, value: &T) -> Option<&T> {
+        let mut current_tree = &self.root;
+
+        while let Some(current_node) = current_tree {
+            match current_node.value.cmp(&value) {
+                Ordering::Less => {
+                    current_tree = &current_node.right;
+                }
+                Ordering::Equal => {
+                    return Some(&current_node.as_ref().value);
+                }
+                Ordering::Greater => {
+                    current_tree = &current_node.left;
+                }
+            }
+        }
+        None
+    }
+}
+
+impl<T: Ord> FromIterator<T> for AVLTreeSet<T> {
+    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
+        let mut set = Self::new();
+
+        for i in iter {
+            set.insert(i);
+        }
+
+        set
+    }
+}
+
+/// Iterator
+impl<'a, T: 'a + Ord> AVLTreeSet<T> {
+    fn iter(&'a self) -> AVLTreeSetNodeIter<'a, T> {
+        AVLTreeSetNodeIter {
+            prev_nodes: Vec::new(),
+            current_tree: &self.root,
+        }
+    }
+}
+
+#[derive(Debug)]
+struct AVLTreeSetNodeIter<'a, T: Ord> {
+    prev_nodes: Vec<&'a AVLNode<T>>,
+    current_tree: &'a AVLTree<T>,
+}
+
+impl<'a, T: 'a + Ord> Iterator for AVLTreeSetNodeIter<'a, T> {
+    type Item = &'a AVLNode<T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
Might be better to return the values itself and not the nodes. I wouldn't leak the nodes to the outside API (don't make it pub).